test/Winton.Extensions.Configuration.Consul.Test/ConsulConfigurationProviderTests.cs

```diff
@@ -112,6 +112,23 @@ private async Task ShouldCancelPollingTaskWhenReloading()
                     kv => kv.List("Test", It.IsAny<QueryOptions>(), It.IsAny<CancellationToken>()),
                     Times.Between(expectedKvCalls, expectedKvCalls + 1, Range.Inclusive));
             }
+
+            [Fact]
+            private void ShouldNotThrowOnMultipleDisposeCalls()
+            {
+                _source.ReloadOnChange = true;
+                _source.Optional = true;
+                _kvEndpoint
+                    .Setup(kv => kv.List("Test", It.IsAny<QueryOptions>(), It.IsAny<CancellationToken>()))
+                    .ReturnsAsync(new QueryResult<KVPair[]> { StatusCode = HttpStatusCode.OK });
+
+                _provider.Load();
+                _provider.Dispose();
+
+                var disposeException = Record.Exception(() => _provider.Dispose());
```

Comment:
We prefer it use fluent assertions. So could you instead do:

```
var secondDispose = _provider.Invoking(p => p.Dispose());

secondDispose.Should().NotThrow();
```

Sorry if syntax not quite right, doing this from mobile.

Reply:
Sorry, I haven't used xUnit much so I was going off of what I found online. Fixed 😄